modules/auxiliary/admin/ldap/rbcd.rb

```diff
@@ -0,0 +1,264 @@
+  end
+
+  def get_delegate_from_obj
+    delegate_from = datastore['DELEGATE_FROM']
+    if delegate_from.blank?
+      fail_with(Failure::BadConfig, 'The DELEGATE_FROM option must be specified for this action.')
+    end
+
+    obj = ldap_get("(sAMAccountName=#{delegate_from})", attributes: ['sAMAccountName', 'ObjectSID'])
+    if obj.nil? && !delegate_from.end_with?('$')
+      obj = ldap_get("(sAMAccountName=#{delegate_from}$)", attributes: ['sAMAccountName', 'ObjectSID'])
+    end
+    fail_with(Failure::NotFound, "Failed to find sAMAccountName: #{delegate_from}") unless obj
+
+    obj
+  end
+
+  def ldap_get(filter, attributes: [])
+    raw_obj = @ldap.search(base: @base_dn, filter: filter, attributes: attributes).first
+    return nil unless raw_obj
+
+    obj = {}
+
+    obj['dn'] = raw_obj['dn'].first.to_s
+    unless raw_obj['sAMAccountName'].empty?
+      obj['sAMAccountName'] = raw_obj['sAMAccountName'].first.to_s
+    end
+
+    unless raw_obj['ObjectSid'].empty?
+      obj['ObjectSid'] = Rex::Proto::MsDtyp::MsDtypSid.read(raw_obj['ObjectSid'].first)
+    end
+
+    unless raw_obj[ATTRIBUTE].empty?
+      obj[ATTRIBUTE] = Rex::Proto::MsDtyp::MsDtypSecurityDescriptor.read(raw_obj[ATTRIBUTE].first)
+    end
+
+    obj
+  end
+
+  def run
+    ldap_connect do |ldap|
+      validate_bind_success!(ldap)
+
+      if (@base_dn = datastore['BASE_DN'])
+        print_status("User-specified base DN: #{@base_dn}")
+      else
+        print_status('Discovering base DN automatically')
+
+        unless (@base_dn = discover_base_dn(ldap))
+          print_warning("Couldn't discover base DN!")
+        end
+      end
+      @ldap = ldap
+
+      delegate_to = datastore['DELEGATE_TO']
+      obj = ldap_get("(sAMAccountName=#{delegate_to})", attributes: ['sAMAccountName', 'ObjectSID', ATTRIBUTE])
+      if obj.nil? && !delegate_to.end_with?('$')
+        obj = ldap_get("(sAMAccountName=#{delegate_to}$)", attributes: ['sAMAccountName', 'ObjectSID', ATTRIBUTE])
+      end
+      fail_with(Failure::NotFound, "Failed to find sAMAccountName: #{delegate_to}") unless obj
+
+      send("action_#{action.name.downcase}", obj)
+    end
+  rescue Net::LDAP::Error => e
+    print_error("#{e.class}: #{e.message}")
+  end
+
+  def action_read(obj)
+    security_descriptor = obj[ATTRIBUTE]
+    if security_descriptor.nil?
+      print_status('The msDS-AllowedToActOnBehalfOfOtherIdentity field is empty.')
+      return
+    end
+
+    if security_descriptor.dacl.nil?
+      print_status('The msDS-AllowedToActOnBehalfOfOtherIdentity DACL field is empty.')
+      return
+    end
+
+    print_status('Allowed accounts:')
+    security_descriptor.dacl.aces.each do |ace|
+      account_name = ldap_get("(ObjectSid=#{ace.body.sid})", attributes: ['sAMAccountName'])
+      print_status("  #{account_name['sAMAccountName']} (#{ace.body.sid})")
```

Comment:
Not a blocker: It might be nice to have the access mask details output here, as I had to put a breakpoint in to debug why I was hitting issues locally - but that might be a dev-specific scenario that users won't necessarily need 😄